vic/drivers/classic/src/read_initial_model_state.c

```diff
@@ -688,4 +692,90 @@ read_initial_model_state(FILE            *init_state,
             }
         }
     }
+
+    /******Check that soil moisture does not exceed maximum allowed************/
```

Comment:
Problem is that the functions that vic_populate_model_state() calls *are* shared across drivers, but to do so requires looping over the grid cells within the non_classic versions of vic_populate_model_state().